### DISCUSSION TITLE
An exception was thrown during execution that is likely a framework error, rather than an error in user code.

### DISCUSSION THREAD

**Discussion by m-alhajji** [dagster-io/dagster#18850] [Q&A]
Hello all 

I've recently ran into a bug that I am not sure how to debug and would appreciate some guidance on where to start to debug it.

One of my jobs started failing consistently with the following logs:
```
An exception was thrown during execution that is likely a framework error, rather than an error in user code.
KeyError: 'model_user_retention'

Stack Trace:
  File "/usr/local/lib/python3.10/site-packages/dagster/_grpc/impl.py", line 137, in core_execute_run
    yield from execute_run_iterator(
,  File "/usr/local/lib/python3.10/site-packages/dagster/_core/execution/api.py", line 866, in __iter__
    yield from self.execution_context_manager.prepare_context()
,  File "/usr/local/lib/python3.10/site-packages/dagster/_utils/__init__.py", line 491, in generate_setup_events
    obj = next(self.generator)
,  File "/usr/local/lib/python3.10/site-packages/dagster/_core/execution/context_creation_job.py", line 291, in orchestration_context_event_generator
    context_creation_data = create_context_creation_data(
,  File "/usr/local/lib/python3.10/site-packages/dagster/_core/execution/context_creation_job.py", line 116, in create_context_creation_data
    resource_keys_to_init=get_required_resource_keys_to_init(execution_plan, job_def),
,  File "/usr/local/lib/python3.10/site-packages/dagster/_core/execution/resources_init.py", line 375, in get_required_resource_keys_to_init
    get_required_resource_keys_for_step(job_def, step, execution_plan)
,  File "/usr/local/lib/python3.10/site-packages/dagster/_core/execution/resources_init.py", line 408, in get_required_resource_keys_for_step
    input_def = node_def.input_def_named(step_input.name)
,  File "/usr/local/lib/python3.10/site-packages/dagster/_core/definitions/node_definition.py", line 143, in input_def_named
    return self._input_dict[name]
```

Any thoughts on how to debug this issue?

**Comment by kylemcleland** [comment]
I'm having the same issue with a similar traceback. Were you able to find a resolution in the meantime?

**Reply by m-alhajji** [reply]
Haven't really found a good way to solve it yet tbh. My current workaround was to create another Dagster job. would be interested to hear your workaround

_I know it's a bad workaround since that doesn't solve my problem but I couldn't find a good solution tbh/don't know what is the root cause of this_

**Reply by smackesey** [reply]
`model_user_retention` is not a string that appears anywhere in the Dagster codebase. Looks like this is an issue with resolution of an input named `model_user_retention` somewhere in your codebase, though we should be catching the issue earlier.

Is `model_user_retention` specified as an input to an asset/op but perhaps not provided in the graph composition phase?

**Reply by m-alhajji** [reply]
yeah model_user_retention is the name of the table. The job materializes this and other dbt assets. I've had this issue as well in another job that materializes all my assets but only if I run it ad-hoc with config full refresh. 

I define all my jobs similarly and they work ok. This job that started failing used to work before and it suddenly started and giving this error. I went back to my git history and couldn't find anything that triggered it. 

it's such a weird issued I don't even know where to start on debugging it. The traceback I sent is all I got so far

**Comment by m-alhajji** [comment]
found a better workaround, re-deploy dagster
